test/builder/auto_mocks_test.dart

```diff
@@ -3553,6 +3553,64 @@ void main() {
         expect(mocksContent, contains('class MockBaz extends _i1.Mock'));
         expect(mocksContent, contains('implements _i2.Baz'));
       });
+
+      test('when its a type parameter of function which returns another type',
+          () async {
+        final mocksContent = await buildWithNonNullable({
+          ...annotationsAsset,
+          'foo|lib/foo.dart': dedent(r'''
+            class Bar {}
+            typedef CreateBar = Bar Function();
+
+            class BaseFoo<T> {
+              BaseFoo(this.t);
+              final T t;
+            }
+
+            class Foo extends BaseFoo<CreateBar> {
+              Foo() : super(() => 1);
+            }
+          '''),
+          'foo|test/foo_test.dart': '''
+            import 'package:foo/foo.dart';
+            import 'package:mockito/annotations.dart';
+
+            @GenerateMocks([Foo])
+            void main() {}
+          '''
+        });
+
+        expect(mocksContent, contains('class MockFoo extends _i1.Mock'));
+        expect(mocksContent, contains('implements _i2.Foo'));
+      });
+      test('when its a duplicate type parameter', () async {
+        final mocksContent = await buildWithNonNullable({
+          ...annotationsAsset,
+          'foo|lib/foo.dart': dedent(r'''
+            class Bar {}
+            typedef BarDef = int Function();
+            typedef BarDef2 = int Function();
+            class BaseFoo<T,P> {
+              BaseFoo(this.t1, this.t2);
+              final T t1;
+              final P t2;
+            }
+            class Foo extends BaseFoo<BarDef, BarDef2> {
+              Foo() : super(() => 1, () => 2);
+            }
+          '''),
+          'foo|test/foo_test.dart': '''
+            import 'package:foo/foo.dart';
+            import 'package:mockito/annotations.dart';
+
+            @GenerateMocks([Foo])
+            void main() {}
+          '''
+        });
+
+        expect(mocksContent, contains('class MockFoo extends _i1.Mock'));
```

Comment:
I don't see any assertions here related to `BarDef` or `BarDef2`.

Reply:
before this pr, the above test would throw an exception that `BarDef2` is not found, so the above is sufficient.
I'll add a small test to if you want me too.
```dart
expect(mocksContent, contains('_i2.BarDef get t1'));
expect(mocksContent, contains('_i2.BarDef2 get t2'));
```